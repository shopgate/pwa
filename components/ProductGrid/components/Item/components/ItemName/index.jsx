import React, { Fragment, PureComponent } from 'react';
import PropTypes from 'prop-types';
import Portal from '@shopgate/pwa-common/components/Portal';
import {
  PRODUCT_ITEM_NAME,
  PRODUCT_ITEM_NAME_AFTER,
  PRODUCT_ITEM_NAME_BEFORE,
} from '@shopgate/pwa-common-commerce/category/constants/Portals';
import Ellipsis from '@shopgate/pwa-common/components/Ellipsis';
import styles from './style';

/**
 * The item name component.
 */
class ItemName extends PureComponent {
  static propTypes = {
    productId: PropTypes.string.isRequired,
    display: PropTypes.shape(),
    name: PropTypes.string,
  };

  static defaultProps = {
    display: null,
    name: null,
  };

  /**
   * @returns {JSX}
   */
  render() {
    const { display, productId, name } = this.props;

    if (display && !display.name) {
      return null;
    }

    const props = { productId };

    return (
      <Fragment>
        <Portal name={PRODUCT_ITEM_NAME_BEFORE} props={props} />
        <Portal name={PRODUCT_ITEM_NAME} props={props}>
          <div className={styles.title} data-test-id={`Productname: ${name}`}>
            <Ellipsis>{name}</Ellipsis>
          </div>
        </Portal>
        <Portal name={PRODUCT_ITEM_NAME_AFTER} props={props} />
      </Fragment>
    );
  }
}

export default ItemName;
