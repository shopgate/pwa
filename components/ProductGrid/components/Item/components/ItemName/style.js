import { css } from 'glamor';
import { responsiveMediaQuery } from '@shopgate/engage/styles';

export default css({
  fontWeight: '500',
  lineHeight: 1.15,
  marginTop: 1,
  wordBreak: ['keep-all', 'break-word'],
  hyphens: 'auto',
  [responsiveMediaQuery('>xs', { webOnly: true })]: {
    fontSize: '1.125rem',
    marginTop: 8,
  },
  [responsiveMediaQuery('>sm', { webOnly: true })]: {
    fontSize: '1.25rem',
    marginTop: 8,
  },
});
