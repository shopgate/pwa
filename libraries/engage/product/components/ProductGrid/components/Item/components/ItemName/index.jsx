import React, { memo, useMemo } from 'react';
import PropTypes from 'prop-types';
import { PRODUCT_ITEM_NAME } from '@shopgate/engage/category/constants';
import { ProductName } from '@shopgate/engage/product';
import { makeStyles } from '@shopgate/engage/styles';

const useStyles = makeStyles()({
  root: {
    fontWeight: '500',
    lineHeight: 1.15,
    marginTop: 1,
    wordBreak: ['keep-all', 'break-word'],
    hyphens: 'auto',
  },
});

/**
 * The item name component.
 * @param {Object} props The component props.
 * @returns {JSX.Element|null}
 */
const ItemName = ({
  display,
  productId,
  name,
}) => {
  const { classes, cx } = useStyles();
  const portalProps = useMemo(() => ({
    productId,
    display,
  }), [display, productId]);

  if (display && !display.name) {
    return null;
  }

  return (
    <ProductName
      name={name}
      className={cx(classes.root, 'theme__product-grid__item__item-name')}
      portalName={PRODUCT_ITEM_NAME}
      portalProps={portalProps}
      testId={`Productname: ${name}`}
    />
  );
};

ItemName.propTypes = {
  productId: PropTypes.string.isRequired,
  display: PropTypes.shape(),
  name: PropTypes.string,
};

ItemName.defaultProps = {
  display: null,
  name: null,
};

export default memo(ItemName);
