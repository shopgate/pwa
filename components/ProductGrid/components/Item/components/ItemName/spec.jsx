import React from 'react';
import { shallow } from 'enzyme';
import ItemName from './index';

const props = {
  productId: '1234',
  name: 'Foo',
};

const display = {
  name: false,
};

describe('<ItemName />', () => {
  it('should render with minimal props', () => {
    const wrapper = shallow(<ItemName {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render with display props set', () => {
    const wrapper = shallow(<ItemName {...props} display={display} />);
    expect(wrapper).toBeEmptyRender();
  });
});
