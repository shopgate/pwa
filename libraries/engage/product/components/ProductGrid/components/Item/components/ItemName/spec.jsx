import React from 'react';
import { shallow } from 'enzyme';
import ItemName from './index';

jest.mock('@shopgate/engage/product', () => ({
  ProductName: () => null,
}));

const props = {
  productId: '1234',
  name: 'Foo',
};

describe('<ItemName />', () => {
  it('should render with minimal props', () => {
    const wrapper = shallow(<ItemName {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
