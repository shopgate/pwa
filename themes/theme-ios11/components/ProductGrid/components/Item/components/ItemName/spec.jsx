import React from 'react';
import { shallow } from 'enzyme';
import ItemName from './index';

jest.mock('@shopgate/engage/product', () => ({
  ProductName: () => null,
}));

const props = {
  productId: '1234',
  name: 'Foo',
};

const display = {
  name: false,
};

describe('<ItemName />', () => {
  it('should render with minimal props', () => {
    const wrapper = shallow(<ItemName {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render with display props set', () => {
    const wrapper = shallow(<ItemName {...props} display={display} />);
    expect(wrapper).toBeEmptyRender();
  });
});
