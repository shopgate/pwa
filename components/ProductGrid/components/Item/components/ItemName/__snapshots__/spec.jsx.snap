// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemName /> should render with minimal props 1`] = `
<Fragment>
  <Portal
    name="product-item.name.before"
    props={
      Object {
        "productId": "1234",
      }
    }
  />
  <Portal
    name="product-item.name"
    props={
      Object {
        "productId": "1234",
      }
    }
  >
    <div
      aria-label="Foo."
      className={
        Object {
          "data-css-5rrf15": "",
        }
      }
      data-test-id="Productname: Foo"
    >
      <Ellipsis
        className=""
        ellipsis="..."
        rows={3}
      >
        Foo
      </Ellipsis>
    </div>
  </Portal>
  <Portal
    name="product-item.name.after"
    props={
      Object {
        "productId": "1234",
      }
    }
  />
</Fragment>
`;
